internal/cloud/atlas/client.go

```diff
@@ -49,8 +50,16 @@ type client struct {
 	transport *digest.Transport
 }
 
-func (c *client) do(method, path string, options api.RequestOptions) (*http.Response, error) {
-	req, reqErr := http.NewRequest(method, c.baseURL+path, options.Body)
+func (c *client) doURL(method, path string, options api.RequestOptions) (*http.Response, error) {
+	return c.do(method, path, options)
+}
+
+func (c *client) doBaseURL(method, path string, options api.RequestOptions) (*http.Response, error) {
```

Comment:
...instead you could keep `c.do` as is and then make this `c.doWithBaseURL` (although the diff of this will be unfortunately larger than probably need be, if we were ok with less-than-optimal names)

open to ideas here, i'd like to:
a) keep the names obvious
b) keep the diff intentional

but there's going to be some amount of compromise you need with this, somewhere